hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/hudi/IncrementalRelation.scala

```diff
@@ -89,8 +89,13 @@ class IncrementalRelation(val sqlContext: SQLContext,
     } else {
       schemaResolver.getTableAvroSchemaWithoutMetadataFields()
     }
-    val dataSchema = AvroConversionUtils.convertAvroSchemaToStructType(tableSchema)
-    StructType(skeletonSchema.fields ++ dataSchema.fields)
+    if (tableSchema == null) {
+      // if there is only one commit in the table and is an empty commit without schema, return empty RDD here
+      null
```

Comment:
For some reason scala was not accepting null values to be returned, used `StructType(Nil)` and made the check for usedSchema accordingly.